astropy/table/array.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+
+class ArraySlot(object):
+    __lt__ = lambda x, y: x.key < y.key
+    __le__ = lambda x, y: x.key <= y.key
+    __eq__ = lambda x, y: x.key == y.key
+    __ge__ = lambda x, y: x.key >= y.key
+    __gt__ = lambda x, y: x.key > y.key
+    __ne__ = lambda x, y: x.key != y.key
+
+    def __init__(self, key, val):
+        self.key = key
+        self.data = val
+    def __repr__(self):
+        return str((self.key, self.data))
+    def __str__(self):
+        return repr(self)
+
+class SortedArray(object):
+    '''
+    Implements a sorted array container using
+    a numpy structured array.
+    '''
+
+    def __init__(self, lines, col_dtypes=None, data=None):
```

Comment:
Ah, thanks for the catch--this made more sense previously when `data` was a structured array, but I forgot about it when I turned `data` into a list of columns. It really should be `self.length = len(data[0])` (the number of elements in each column), and by the way `data` is only used for slicing. I'll change `lines` to `key_cols_table` as well since that seems like a better name.